pkg/detectors/github/github.go

```diff
@@ -94,10 +97,28 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 							s1.ExtraData["username"] = userResponse.Login
 							s1.ExtraData["url"] = userResponse.UserURL
 							s1.ExtraData["account_type"] = userResponse.Type
-							s1.ExtraData["site_admin"] = fmt.Sprintf("%t", userResponse.SiteAdmin)
-							s1.ExtraData["name"] = userResponse.Name
-							s1.ExtraData["company"] = userResponse.Company
-							s1.ExtraData["scopes"] = res.Header.Get("X-OAuth-Scopes")
+							if userResponse.SiteAdmin {
+								s1.ExtraData["site_admin"] = "true"
+							}
+							if userResponse.Name != "" {
+								s1.ExtraData["name"] = userResponse.Name
+							}
+							if userResponse.Company != "" {
+								s1.ExtraData["company"] = userResponse.Company
+							}
+							if userResponse.LdapDN != "" {
+								s1.ExtraData["ldap_dn"] = userResponse.LdapDN
+							}
+
+							// GitHub does not seem to consistently return this header.
+							scopes := res.Header.Get("X-OAuth-Scopes")
+							if scopes != "" {
+								s1.ExtraData["scopes"] = scopes
+							}
```

Comment:
Based on my testing, this header doesn't seem to be included when the new `github_pat_x` tokens are used. I don't know if this is intentional.